var/spack/repos/builtin/packages/py-scikit-learn/package.py

```diff
@@ -71,34 +72,20 @@ class PyScikitLearn(PythonPackage):
     # See https://github.com/scikit-learn/scikit-learn/issues/14332
     conflicts('~openmp', when='@:999', msg='Only master supports ~openmp')
 
-    def setup_environment(self, spack_env, run_env):
+    def setup_build_environment(self, env):
         # https://scikit-learn.org/stable/developers/advanced_installation.html#building-from-source
         if self.spec.satisfies('~openmp'):
-            spack_env.set('SKLEARN_NO_OPENMP', 'True')
+            env.set('SKLEARN_NO_OPENMP', 'True')
         # https://scikit-learn.org/stable/developers/advanced_installation.html#mac-osx
         elif self.spec.satisfies('@0.21: %clang platform=darwin +openmp'):
-            spack_env.append_flags(
+            env.append_flags(
                 'CPPFLAGS', '-Xpreprocessor -fopenmp')
-            spack_env.append_flags(
+            env.append_flags(
                 'CFLAGS', self.spec['llvm-openmp'].headers.include_flags)
-            spack_env.append_flags(
-                'CXXFLAGS',
-                self.spec['llvm-openmp'].headers.include_flags)
-            spack_env.append_flags(
+            env.append_flags(
+                'CXXFLAGS', self.spec['llvm-openmp'].headers.include_flags)
+            env.append_flags(
                 'LDFLAGS', self.spec['llvm-openmp'].libs.ld_flags)
-            spack_env.append_flags(
-                'DYLD_LIBRARY_PATH',
-                self.spec['llvm-openmp'].libs.directories[0])
-
-            run_env.append_flags(
-                'DYLD_LIBRARY_PATH',
-                self.spec['llvm-openmp'].libs.directories[0])
-
-    def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
-        if self.spec.satisfies('@0.21: %clang platform=darwin +openmp'):
-            spack_env.append_flags(
-                'DYLD_LIBRARY_PATH',
-                self.spec['llvm-openmp'].libs.directories[0])
```

Comment:
Pretty sure these `DYLD_LIBRARY_PATH` settings aren't required, as RPATH seems to be working correctly, but let me confirm.

Reply:
Yeah, this seems to be working. This should be safe to merge.